src/ensembl/src/content/app/entity-viewer/gene-view/components/gene-view-sidebar/overview/MainAccordion.tsx

```diff
@@ -31,41 +32,60 @@ import { ReactComponent as DownloadButton } from 'static/img/launchbar/custom-do
 import Checkbox from 'src/shared/components/checkbox/Checkbox';
 
 import { Status } from 'src/shared/types/status';
-import JSONValue from 'src/shared/types/JSON';
-import { EntityViewerSidebarPayload } from 'src/content/app/entity-viewer/state/sidebar/entityViewerSidebarState';
+import {
+  EntityViewerSidebarPayload,
+  EntityViewerSidebarUIState
+} from 'src/content/app/entity-viewer/state/sidebar/entityViewerSidebarState';
 
 import styles from './GeneOverview.scss';
 
 type Props = {
   sidebarPayload: EntityViewerSidebarPayload | null;
-  sidebarUIState: { [key: string]: JSONValue } | null;
-  updateEntityUI: (uIstate: { [key: string]: JSONValue }) => void;
+  sidebarUIState: EntityViewerSidebarUIState | null;
+  updateEntityUI: (uIstate: Partial<EntityViewerSidebarUIState>) => void;
 };
 
+export type AccordionSectionID = 'function' | 'sequence' | 'other_data_sets';
+
 // TODO: Remove me once instant download component is available
 const mockOnClick = noop;
 
 const MainAccordion = (props: Props) => {
-  if (!props.sidebarPayload) {
+  const { sidebarPayload, sidebarUIState } = props;
+
+  if (!sidebarPayload) {
     return null;
   }
-  const { gene } = props.sidebarPayload;
-  const expandedPanels = props.sidebarUIState?.mainAccordion
-    ?.expandedPanels as string[];
 
-  const onChange = (expandedPanels: (string | number)[] = []) => {
+  const { gene } = sidebarPayload;
+  const hasFunctionDescription = Boolean(gene.function?.description);
+
+  const expandedPanels =
+    (sidebarUIState?.mainAccordion?.expandedPanels as AccordionSectionID[]) ||
```

Comment:
Why doesn't `sidebarUIState?.mainAccordion?.expandedPanels` already know about itself that it is of type AccordionSectionID[]? Why isn't this type in our redux state?

Reply:
The type is there in our redux. I missed to remove the casting.